application.go

```diff
@@ -109,6 +129,28 @@ type DeleteAppOpts struct {
 	Force bool `url:"force,omitempty"`
 }
 
+// Diego ======
+
+// SetIPAddressPerTask defines that the application will have a IP address defines by a external agent. this
+// configuration is not allowed to be used with Port or PortDefinitions.
+func (r *Application) SetIPAddressPerTask(ipAddressPerTask *IPAddressPerTask) *Application {
+
+	r.Ports = make([]int, 0)
+	r.EmptyPortDefinitions()
+	r.IPAddressPerTask = ipAddressPerTask
+
+	return r
+}
+
+// EmptyIPAddressPerTask explicitly empties port definitions -- use this if you need to empty
+// IP address per task of an application that already has some definitions set (setting port definitions to nil will
+// keep the current value)
+func (r *Application) EmptyIPAddressPerTask() *Application {
+	r.IPAddressPerTask = nil
```

Comment:
The idea of the `Empty*` methods is that we support clearing a value previously set. However, setting it to nil would simply not pass the field in the resulting JSON due to our usage of the omitempty tag. What we would need to do instead is pass the address of an empty `IPAddressPerTask` struct, i.e., `&IPAddressPerTask{}`.

For the case at hand, however, it doesn't have the expected effect: According to the documentation, an empty `ipAddress` field yields the default configuration for an enabled IP-per-task setting. The API does not seem to provide a way to disable the feature once enabled.

If you share my observation, then I suggest we remove `EmptyIPAddressPerTask` entirely in order to avoid confusing users into believing they are getting a feature which they don't. (If someone wants to return to the default setting after having set something more specific (like discovery), they can still do so by calling `SetIPAddressPerTask` with an empty struct.)

Apologies for having mislead in the other PR by originally advocating in favor of this method. I only realized now that the field we are dealing with carries different semantics.